src/utils/helpers.js

```diff
@@ -25,3 +25,27 @@ export const setLocalStorage = (
   localStorage.setItem("activeSet", JSON.stringify(activeSet));
   localStorage.setItem("round", JSON.stringify(round));
 };
+
+export const chooseGroups = (groupSet) => {
+  let members = groupSet.values;
+  const size = 3;
+  const groupsChosen = [];
+
+  while (members.length > 0) {
+    let group = [];
+    if (members.length >= size) {
+      while (group.length < size) {
+        let index = Math.floor(Math.random() * members.length);
+        // if (!group.includes(members[index].name)) {
+        group.push(members[index].name);
+        members.splice(index, 1);
+        // }
+      }
+    } else {
+      group = members.map((member) => member.name);
+      members = [];
+    }
+    groupsChosen.push(group);
+  }
+  return groupsChosen;
+};
```

Comment:
```suggestion
export const chooseGroups = (groupSet, size = 3) => {
  const randomMembers = [...groupSet.values].sort(() => 0.5 - Math.random());

  const groups = [];
  randomMembers.forEach((member) => {
    let lastGroup = groups[groups.length - 1];
    if (!lastGroup || lastGroup.length >= size) {
      groups.push([]);
      lastGroup = groups[groups.length - 1];
    }

    lastGroup.push(member.name);
  });

  return groups;
};
```

Or with [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)

```suggestion
export const chooseGroups = (groupSet, size = 3) => {
  const randomMembers = [...groupSet.values].sort(() => 0.5 - Math.random());

  const groups = randomMembers.reduce((groupsAccumulator, member) => {
    let lastGroup = groupsAccumulator[groupsAccumulator.length - 1];
    if (!lastGroup || lastGroup.length >= size) {
      groupsAccumulator.push([]);
      lastGroup = groupsAccumulator[groupsAccumulator.length - 1];
    }

    lastGroup.push(member.name);
    return groupsAccumulator;
  }, []);

  return groups;
};
```